pages/api/v1/secret/index.ts

```diff
@@ -0,0 +1,76 @@
+import { NextRequest, NextResponse } from "next/server";
+import { Redis } from "@upstash/redis";
+import { generateId } from "pkg/id";
+import { z } from "zod";
+
+export const requestValidation = z.object({
+  // ttl in seconds
+  // defaults to 30 days
+  // not more than 1 year
+  ttl: z
+    .string()
+    .nullable()
+    .transform((v) => (v ? parseInt(v, 10) : 43260))
+    .refine((v) => v > 0 && v <= 30758400, "ttl must be between 1 and 30758400 seconds"),
+
+  // number of reads before deletion
+  reads: z
+    .string()
+    .nullable()
+    .transform((v) => (v ? parseInt(v, 10) : null))
+    .refine((v) => v === null || v > 0, "reads must be greater than 0"),
+  secret: z.string().min(1),
+});
+export const responseValidation = z.union([
+  z.object({
+    data: z.object({
+      id: z.string(),
+    }),
+  }),
+  z.object({
+    error: z.string(),
+  }),
+]);
+
+const redis = Redis.fromEnv();
+
+export default async function handler(req: NextRequest): Promise<NextResponse> {
+  try {
+    if (req.method !== "POST") {
+      return NextResponse.json({ error: "Method Not Allowed" }, { status: 405 });
+    }
+
+    const parsed = requestValidation.safeParse({
+      ttl: req.headers.get("envshare-ttl"),
+      reads: req.headers.get("envshare-reads"),
+      secret: await req.text(),
+    });
+    if (!parsed.success) {
+      return NextResponse.json({ error: JSON.stringify(JSON.parse(parsed.error.message), null, 2) }, { status: 400 });
```

Comment:
Could we return this isntead?  `{ error: JSON.parse(parsed.eror.message)` 
or prettify it because this results in: 
![image](https://user-images.githubusercontent.com/16673873/213066554-6e68b03c-587e-4930-aac1-9108c21ebb0a.png)

So it can look like this instead:
![image](https://user-images.githubusercontent.com/16673873/213073323-adc66443-c40c-40ef-b011-4c4dcb91d577.png)
![image](https://user-images.githubusercontent.com/16673873/213073477-b5b6a6f2-393f-4e60-b1e9-531256ec1b4b.png)

Could just be on my end though, can you check it out?

Reply:
Yeah this is a problem I always have with zod :D no idea how to fix it right now